tests/v2/validation/configmaps/configmaps_test.go

```diff
@@ -0,0 +1,113 @@
+  require.NotEmptyf(s.T(), clusterName, "Cluster name to install should be set")
+  clusterID, err := clusters.GetClusterIDByName(s.client, clusterName)
+  require.NoError(s.T(), err, "Error getting cluster ID")
+  s.cluster, err = s.client.Management.Cluster.ByID(clusterID)
+  require.NoError(s.T(), err)
+}
+
+func (s *SteveAPITestSuite) TestConfigMapCreate() {
+  subSession := s.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+  
+  log.Info("Creating a configmap")
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapNamespace := namespace
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, configmapNamespace, annotations, labels, data)
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was created")
+  configmapByID, err := configMapClient.ByID(configmapObj.ID)
+  require.NoError(s.T(), err)
+  assert.Contains(s.T(), configmapByID.JSONResp["id"], configmapName)
+}
+
+func (s *SteveAPITestSuite) TestConfigMapUpdate(){
+  subSession := s.session.NewSession()
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+
+  log.Info(("Creating a configmap"))
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapNamespace := namespace
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, configmapNamespace, annotations, labels, data)
+  require.NoError(s.T(), err)
+
+  log.Info("Updating the configmap")
+  updatedConfigMapObj, err := updateConfigmapAnnotations(*configMapClient, configmapObj, updatedAnnotations)
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was properly updated")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations, "anno2")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations["anno2"], "new automated annotation")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations, "field.cattle.io/description")
+  assert.Contains(s.T(), updatedConfigMapObj.ObjectMeta.Annotations["field.cattle.io/description"], "new automated configmap description")
+}
+
+func (s *SteveAPITestSuite) TestConfigmapDelete(){
+  subSession := s.session.NewSession()
+  s.session = subSession
+  defer subSession.Cleanup()
+
+  configMapClient := s.client.Steve.SteveType(configmaps.ConfigMapSteveType)
+
+  log.Info("Creating a configmap")
+  configmapName := namegenerator.AppendRandomString("steve-configmap")
+  configmapNamespace := namespace
+  configmapObj, err := createConfigmap(*configMapClient, configmapName, configmapNamespace, annotations, labels, data)
+  require.NoError(s.T(), err)
+
+
+  log.Info("Deleting the configmap")
+  err = configMapClient.Delete(&configmapObj)
+  require.NoError(s.T(), err)
+
+  log.Info("Validating configmap was deleted")
+  configmapByID, err := configMapClient.ByID(configmapObj.ID)
+  require.Error(s.T(), err)
```

Comment:
can we verify the error that we get is `not found` Error here as an additional check.